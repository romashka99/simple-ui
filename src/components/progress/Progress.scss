@import '../../styles/colors.scss';
@import '../../styles/main.scss';
@import '../../styles/style.scss';

.progress {
	display: flex;
	align-items: center;

	.progress--label {
		display: flex;
		align-items: center;
		padding-left: 8px;
		color: $color-grey;
		font-size: 16px;
		line-height: 24px;
	}

	.progress--inner {
		position: relative;
		height: 5px;
		width: 100%;
		background-color: $color-mid-light-grey;
		border-radius: 5px;
	}

	.progress--filled {
		position: absolute;
		height: 5px;
		border-radius: 5px;
		background-color: $color-grey;
	}
}
