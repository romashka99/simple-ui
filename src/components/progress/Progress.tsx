import ErrorIcon from 'components/icons/ErrorIcon';
import SuccessIcon from 'components/icons/SuccessIcon';
import React from 'react';
import './Progress.scss';

export type StatusType = 'success' | 'error' | 'active';

interface IProgressProps extends React.AriaAttributes {
	value: number;
	showInfo?: boolean;
	status?: StatusType;
	className?: string;
}

export default function Progress({
	value,
	showInfo = true,
	status,
	className,
}: IProgressProps) {
	const classes = `progress${className ? ' ' + className : ''}`;

	return (
		<div data-testid='progress' className={classes}>
			<div data-testid='progress-inner' className='progress--inner'>
				<div style={{ width: `${value}%` }} className='progress--filled'></div>
			</div>
			{showInfo && (
				<label data-testid='progress-label' className='progress--label'>
					{status === 'error' ? (
						<ErrorIcon />
					) : status === 'success' ? (
						<SuccessIcon />
					) : (
						`${value}%`
					)}
				</label>
			)}
		</div>
	);
}
