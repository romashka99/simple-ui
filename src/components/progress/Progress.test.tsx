import { render, screen, within } from '@testing-library/react';
import Progress from '.';

describe('Test Progress', () => {
	test('without attr', () => {
		render(<Progress value={10} />);
		const progress = screen.getByTestId('progress');
		const inner = within(progress).getByTestId('progress-inner');
		const label = within(progress).getByTestId('progress-label');
		expect(inner).not.toBeNull();
		expect(label).not.toBeNull();
		expect(progress.classList.length).toBe(1);
		expect(progress.classList.contains('progress')).toBe(true);
		expect(progress).toMatchSnapshot();
	});
});
