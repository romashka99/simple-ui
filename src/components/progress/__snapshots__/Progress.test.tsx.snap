// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Progress without attr 1`] = `
<div
  class="progress"
  data-testid="progress"
>
  <div
    class="progress--inner"
    data-testid="progress-inner"
  >
    <div
      class="progress--filled"
      style="width: 10%;"
    />
  </div>
  <label
    class="progress--label"
    data-testid="progress-label"
  >
    10%
  </label>
</div>
`;
